<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE properties SYSTEM "cpf.dtd">
<properties>
<simple name="path" type="string"><value>$(find amigo_spindle_controller)/lib/orocos</value></simple>
<simple name="path" type="string"><value>$(find soem_master)/lib/orocos</value></simple>
<simple name="path" type="string"><value>$(find soem_beckhoff_drivers)/lib/orocos</value></simple>
<simple name="path" type="string"><value>$(find rtt_ros_integration)/lib/orocos</value></simple>
<simple name="path" type="string"><value>$(find rtt_ros_integration_std_msgs)/lib/orocos</value></simple>
<simple name="path" type="string"><value>$(find rtt_ros_integration_amigo_msgs)/lib/orocos</value></simple>
<simple name="path" type="string"><value>$(find ocl)/install/lib/orocos</value></simple>
<simple name="path" type="string"><value>$(find rtt)/install/lib/orocos</value></simple>

<struct name="ROSConSpindleSetpoint" type="ConnPolicy">
  <simple name="type" type="short"><value>0</value></simple>
  <simple name="size" type="short"><value>1</value></simple>
  <simple name="transport" type="short"><value>3</value></simple>
  <simple name="name_id" type="string"><value>/spindle_setpoint</value></simple>
</struct>

<struct name="ROSConSpindlePos" type="ConnPolicy">
  <simple name="type" type="short"><value>0</value></simple>
  <simple name="size" type="short"><value>1</value></simple>
  <simple name="transport" type="short"><value>3</value></simple>
  <simple name="name_id" type="string"><value>/spindle_pos</value></simple>
</struct>

<struct name="SoemMaster" type="soem_master::SoemMasterComponent">
  <struct name="Activity" type="PeriodicActivity">
    <simple name="Period" type="double"><value>0.004</value></simple>
    <simple name="Priority" type="short"><value>1</value></simple>
    <simple name="Scheduler" type="string"><value>ORO_SCHED_RT</value></simple>
  </struct>
</struct>

<struct name="SpindleController" type="SpindleController">
  <struct name="Activity" type="PeriodicActivity">
    <simple name="Period" type="double"><value>0.004</value></simple>
    <simple name="Priority" type="short"><value>1</value></simple>
    <simple name="Scheduler" type="string"><value>ORO_SCHED_RT</value></simple>
  </struct>
  <struct name="Ports" type="PropertyBag">
    <simple name="spindle_position" type="string"><value> ROSConSpindlePos </value></simple>
    <simple name="spindle_setpoint" type="string"><value> ROSConSpindleSetpoint </value></simple>
  </struct>
  <simple name="LoadProperties" type="string"><value>$(find amigo_spindle_controller)/spindle_parameters.cpf</value></simple>
</struct>


<struct name="Deployer"  type="PropertyBag">
  <simple name="ProgramScript" type="string"><value>$(find amigo_spindle_controller)/startup.ops</value></simple>
  <simple name="AutoStart" type="boolean"><value>1</value></simple>
  <struct name="Activity" type="Activity">
    <simple name="Period" type="double"><value>0.1</value></simple>
    <simple name="Priority" type="short"><value>1</value></simple>
    <simple name="Scheduler" type="string"><value>ORO_SCHED_OTHER</value></simple>
  </struct>
</struct>


<struct name="Reporter" type="OCL::FileReporting">
    <!-- Note: Activity may also be non-periodic -->
    <struct name="Activity" type="Activity">
      <simple name="Period" type="double"><value>0.004</value></simple>
      <simple name="Priority" type="short"><value>2</value></simple>
      <simple name="Scheduler" type="string"><value>ORO_SCHED_OTHER</value></simple>
    </struct>
    <simple name="AutoConf" type="boolean"><value>0</value></simple>
    <simple name="AutoStart" type="boolean"><value>0</value></simple>
    <simple name="AutoSave" type="boolean"><value>0</value></simple>
    <simple name="LoadProperties" type="string"><value>$(find amigo_spindle_controller)/reporting.cpf</value></simple>
    <!-- List all peers (uni-directional) -->
    <struct name="Peers" type="PropertyBag">
      <simple type="string"><value>SpindleController</value></simple>
    </struct>
</struct>

</properties>

